internal/rootcoord/dml_channels.go

```diff
@@ -17,42 +17,137 @@
 package rootcoord
 
 import (
+	"container/heap"
 	"context"
 	"fmt"
 	"sync"
 
 	"github.com/milvus-io/milvus/internal/metrics"
 
-	"go.uber.org/atomic"
 	"go.uber.org/zap"
 
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/mq/msgstream"
 )
 
 type dmlMsgStream struct {
-	ms     msgstream.MsgStream
-	mutex  sync.RWMutex
-	refcnt int64
+	ms       msgstream.MsgStream
+	mutex    sync.RWMutex
+	chanName string
+	refcnt   int64 // current in use count
+	used     int64 // total used counter in current run, not stored in meta so meant to be inaccurate
+	idx      int64 // idx for name
+	pos      int   // position in the heap slice
+}
+
+// RefCnt returns refcnt with mutex protection.
+func (dms *dmlMsgStream) RefCnt() int64 {
+	dms.mutex.RLock()
+	defer dms.mutex.RUnlock()
+	return dms.refcnt
+}
+
+// RefCnt returns refcnt with mutex protection.
+func (dms *dmlMsgStream) Used() int64 {
+	dms.mutex.RLock()
+	defer dms.mutex.RUnlock()
+	return dms.used
+}
+
+// AddUsage increases refcnt and used.
+func (dms *dmlMsgStream) AddUsage() {
```

Comment:
in your comments, you said "increase refcnt and used",
but in the code, only refcnt is increased.

Reply:
Nice catch. The behavior was changed but the comment was not changed.

Reply:
Fixed